controller/master_controller.py

```
@@ -25,6 +24,83 @@ class MasterClient:
+            else:
+                MasterModel(**data.__dict__)
+        except ValidationError as e:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail={"error": "Validation Error", "details": e.errors()}
+            )
+
+    @staticmethod
+    def create_master_entries(table_name: str, data: CreateMasterModel):
+        if table_name not in MasterClient.table:
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="Invalid table name.")
+        try:
+            MasterClient.validate_master_data(data)
+            master_data = MasterClient.fetch_data(table_name, data)
+            if master_data and not master_data[0]['is_active']:
+                return MasterClient.update_master(master_data[0]['id'], data.is_active, table_name)
+            elif master_data:
+                raise Exception("Data already exists")
+            else:
+                return MasterClient.create_master_data(table_name, data)
+        except Exception as e:
+            logger.error(f"Error creating master entry: {e}")
+            return JSONResponse(
+                content={"error": str(e), "details": "Unable to fetch the details."},
+                status_code=status.HTTP_400_BAD_REQUEST
+            ) 
+        
+    @staticmethod
+    def update_master(resource_id, active, table_name):
+        try:
+            master_data = API.make_request(
+                method="PATCH",
+                endpoint=f"/fhir/{MasterClient.table[table_name]}/{resource_id}",
+                json={"is_active": active}
+            )
+            if master_data.status_code == 200:
+                values = master_data.json()
+                return values
+            else:
+                logger.error(table_name)
+                logger.error(master_data.status_code)
+                logger.error(master_data.json())
+                raise Exception("Unable to create")
```

can you raise the proper exception? update method right?

---

If possible add the table name as well, unable to update the which master data/ like / location/lab/ etc...

---

added for update method and added the table